netscaler/netscaler_resource.go

```diff
@@ -124,15 +163,27 @@ func (c *NitroClient) doHTTPRequest(method string, url string, bytes *bytes.Buff
 		return []byte{}, err
 	}
 	log.Println("[DEBUG] go-nitro: response Status:", resp.Status)
-	return respHandler(resp)
+	body, err := respHandler(resp)
+	// Clear sessionid in case of session-expiry
+	if resp.Status == "401 Unauthorized" {
+		var data map[string]interface{}
+		err2 := json.Unmarshal(body, &data)
+		if err2 == nil {
+			data["errorcode"] = int(data["errorcode"].(float64))
```

Comment:
this is unusual. Why not create a new var instead of overwriting the unmarshaled map?